drivers/pds/targetcluster/targetcluster.go

```diff
@@ -80,8 +82,12 @@ func (tc *TargetCluster) GetDeploymentTargetID(clusterID, tenantID string) (stri
 			deploymentTargetID = targetClusters[i].GetId()
 			log.Infof("deploymentTargetID %v", deploymentTargetID)
 			log.InfoD("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
+			targetClusterStatus = targetClusters[i].GetStatus()
 		}
 	}
+	if targetClusterStatus != "healthy" {
+		return "Target Cluster is not in healthy state, hence exiting the testcase execution", fmt.Errorf("error occured is: %v", err)
```

Comment:
@jkagliwal-px the return should be in the form of nil,<error> and not <string>,<error>. Also you can embed the error message in returned error itself.